src/Middleware/HttpLogging/src/FileLoggerProcessor.cs

```diff
@@ -140,6 +157,23 @@ private async Task WriteMessagesAsync(List<string> messages, CancellationToken c
             // Files are written up to _maxFileSize before rolling to a new file
             DateTime today = DateTime.Now;
             var fullName = GetFullName(today);
+            // Don't write to an incomplete file left around by a previous FileLoggerProcessor
+            if (_firstFile)
+            {
+                while (File.Exists(fullName))
+                {
+                    _fileNumber++;
+                    if (_fileNumber >= W3CLoggerOptions.MaxFileCount)
+                    {
+                        _maxFilesReached = true;
+                        // Return early if log directory is already full
+                        Log.MaxFilesReached(_logger, new ApplicationException());
```

Comment:
I know this was here before, but does passing in an ApplicationException() really give us anything? Can we just leave the exception null?

Reply:
@BrennanConroy beat me to it!